Addressed issue with inconsistent `EnzoComputeTemperature` units…

mabruzzo opened the pull request:
Addressed issue where `EnzoComputeTemperature` returned the `"temperature"` in different units based on whether we used Grackle or not (when using Grackle, the output was in Kelvin). @brittonsmith and I talked about it, and we decided to track the temperature field in units of Kelvin (since temperature units are kind of a nebulous).

I modified `EnzoMethodTurbulence`, accordingly. Other people should speak up if they disagree. I took a quick glance at `EnzoInitialIsolatedGalaxy` and `EnzoInitialGrackle`, and I don't think they require any changes.

I'm tempted to remove the `Units::temperature()` method and rename `EnzoUnits::temperature()` so it's called something like `EnzoUnits::temperature_in_energy_units()`. I would appreciate feedback about this latter point (maybe we turn this into an Issue).

mabruzzo commented on src/Enzo/enzo_EnzoMethodTurbulence.cpp:
```diff
@@ -134,6 +134,13 @@ void EnzoMethodTurbulence::compute ( Block * block) throw()
   field.dimensions (0,&mx,&my,&mz);
   const int rank = ((mz == 1) ? ((my == 1) ? 1 : 2) : 3);
 
+  // the temperature field is in units of K. we need to convert it to units
+  // of specific energy (or velocity_units^2). Some people might refer to
+  // this as "temperature units".
+  EnzoUnits* enzo_units = enzo::units();
+  enzo_float temp_to_energy_u =
+    cello::mass_hydrogen*std::pow(enzo_units->velocity(),2)/cello::kboltz;
+
```
It's a conversion to energy units in code units. I'll update the comment accordingly

stefanarridge replied:
Are you sure about this? I might be missing something, but I'm fairly sure than `enzo_units->velocity()`is unit velocity divided by `cm/s`. So, if the unit velocity is `km/s`, `enzo_units->velocity()` will have the value `1000`. 

Also, this is a minor thing, but (or so I'm told), when you square a variable `x`, its much more efficient to do `x * x` than `pow(x,2)`.

gregbryan reviewed (approved):
This seems good to me.  In general, I am strongly in favor of renaming to make things maximally clear and minimize the chance for misinterpreting "temperature". Different fields have different shorthands.

mabruzzo commented:
So, I fixed the bug that Stephan pointed out.

I also renamed `EnzoUnits::temperature()` -> `EnzoUnits::kelvin_per_energy_units` to try to avoid introducing the  "temperature units" all-together. Along a similar line of thought I ultimately decided to delete the `Units::temperature()` method.

Assuming that all tests pass, this should be good to go
